pkg/ruler/base/ruler.go

```diff
@@ -799,26 +799,69 @@ func RemoveRuleTokenFromGroupName(name string) string {
 
 // GetRules retrieves the running rules from this ruler and all running rulers in the ring if
 // sharding is enabled
-func (r *Ruler) GetRules(ctx context.Context) ([]*GroupStateDesc, error) {
+func (r *Ruler) GetRules(ctx context.Context, req *RulesRequest) ([]*GroupStateDesc, error) {
 	userID, err := tenant.TenantID(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("no user id found in context")
 	}
 
 	if r.cfg.EnableSharding {
-		return r.getShardedRules(ctx, userID)
+		return r.getShardedRules(ctx, userID, req)
 	}
 
-	return r.getLocalRules(userID)
+	return r.getLocalRules(userID, req)
 }
 
-func (r *Ruler) getLocalRules(userID string) ([]*GroupStateDesc, error) {
+type StringFilterSet map[string]struct{}
+
+func makeStringFilterSet(values []string) StringFilterSet {
+	set := make(map[string]struct{}, len(values))
+	for _, v := range values {
+		set[v] = struct{}{}
+	}
+	return set
+}
+
+// IsFiltered returns whether to filter the value or not.
+// If the set is empty, then nothing is filtered.
+func (fs StringFilterSet) IsFiltered(val string) bool {
+	if len(fs) == 0 {
+		return false
+	}
+	_, ok := fs[val]
+	return !ok
+}
+
+func (r *Ruler) getLocalRules(userID string, req *RulesRequest) ([]*GroupStateDesc, error) {
+	var (
+		getRecordingRules = true
```

Comment:
I would prefer to create and switch these flags more direct way.

1. start with both false.
2. case AlertingRule => `alertRule=true`
3. case RecordingRule => `recordingRule=true`
4. case AnyRule => `alerRule=true; recordingRule=true`

This way it's more intuitive IMHO.

Reply:
Also one other question. Do you think we also should support `exclude_alerts=<bool>`, just like prometheus rules endpoint does.
https://prometheus.io/docs/prometheus/latest/querying/api/#rules

Reply:
i'll add `exclude_alerts` in a follow-up